packages/core/src/plugins/use-schema.ts

```diff
@@ -9,20 +10,47 @@ export const useSchema = (schema: GraphQLSchema): Plugin => {
   };
 };
 
-export const useLazyLoadedSchema = (schemaLoader: (context: Maybe<DefaultContext>) => GraphQLSchema): Plugin => {
+export const useLazyLoadedSchema = (schemaLoader: (context: Maybe<DefaultContext>) => PromiseOrValue<GraphQLSchema>): Plugin => {
+  let schemaSet$: PromiseOrValue<void>;
   return {
     onEnveloped({ setSchema, context }) {
-      setSchema(schemaLoader(context));
+      const schema$ = schemaLoader(context);
+      if (isPromise(schema$)) {
+        schemaSet$ = schema$.then(schemaObj => {
+          setSchema(schemaObj);
+        });
+      } else {
+        setSchema(schema$);
+      }
+    },
+    onExecute() {
```

Comment:
So the idea here is to defer execution until the schema is set - But that will still let validation fail 🤔

Reply:
That's why I left this PR draft :D because only if you don't do `validate` you can use promises for `schema`.